### ISSUE TITLE
DSC benchmark sharing

### ISSUE THREAD

**Issue by gaow** [stephenslab/dsc2#89]
@jhsiao999 and I are designing a single cell benchmarking project. A big part of it is to browse and share the benchmark. Now that we have `dscrutils::dsc_query` it is time to bring this back to discussion. 

Currently there are 2 approaches:

1. Thanks to Nan and Wei we have a draft `shinydsc` R package that is flexible enough to work with an earlier version of `dsc-query` syntax. There should not be too much changes to make it work for the new one.
2. `dsc-query` also output Jupyter notebook for each condition of interest, that people can keep exploring inside the notebook, and [publish multiple notebooks as a website using an established procedure](https://github.com/stephenslab/ipynb-website).

Pros  / cons:
1. Shinydsc: people who view DSC benchmark can get more involved in looking at results of **their** interest interactively; this would be a con for Jupyter. Also a shinyapp is always more "shiny" to show off.
2. Jupyter approach: Provide an easy way to summarize explorations of multiple scenarios into a static website that documents everything for others to browse. People can easily see what **we** have explored and highlight. Shinydsc, on the other hand, can save / load configuration files to display what we have explored, but is not convenient if people just want to glance over static results.

Implementaiton-wise:
1. Shinydsc needs a lot more engineering.
2. Juptyer approach may easily be converted to a `workflowr` approach instead, by extending `dscr::utils` to work with `workflowr`.

Looks like perhaps 2 is the way to go in short term, and we just need to figure out a way for people to collectively maintain a `workflowr` site, as a way to "extend" DSC?  @jdblischak ?

**Comment by jdblischak** [comment]
> Looks like perhaps 2 is the way to go in short term, and we just need to figure out a way for people to collectively maintain a workflowr site, as a way to "extend" DSC? @jdblischak ?

@gaow Happy to discuss this sometime. Not clear to me from the above description how tightly linked you want dsc and workflowr to be.

**Comment by gaow** [comment]
@jdblischak Great! Implementation-wise perhaps the only thing on DSC side that ` workflowr` potentially needs to talk to is [just this function](https://github.com/stephenslab/dsc2/blob/master/dscrutils/R/dscquery.R) that Peter recently has developed. But without a context it maybe quite difficult to communicate. I guess the single-cell benchmarking project is the best opportunity to design and test it out. So before we dive to details I guess I'd first work with Joyce to get some prelim result [migrating this benchmark](https://github.com/markrobinsonuzh/conquer) to DSC, then with the benchmark data we harvest we can discuss formally how we want it to be in a presentable format and how `workflowr` can specifically help automating it (or how we design it around `workflowr`)

**Comment by gaow** [comment]
BTW this is the benchmark sharing the project we want to port has provided:

http://imlspenticton.uzh.ch:3838/conquer/

and you can perhaps tell what improvements we can make. The key is to be able to do it in automated and reproducible fashion.

**Comment by jdblischak** [comment]
@gaow OK, so my first impression was that workflowr is a large dependency to add to DSC. If DSC wants to create some custom reports given in an overview of the results, it may make more sense to just generate some HTML directly without involving workflowr.

workflowr is designed to help with the exploratory data analysis phase of a project. The DSC step is analogous to my Snakemake pipeline to process sequencing data. I automate the data processing with Snakemake, and I don't involve workflowr until the data is ready to be imported into R for bespoke data analysis. Once a computational step becomes routine, it can be migrated to the Snakemake pipeline. I envision the same setup for a DSC project. After the pipeline has been executed, the user can start a workflowr analysis to run arbitrary complex queries directly with `dsc_query` and then immediately manipulate and visualize the results.

@stephens999, based on your recent Slack post, I feel like we are the same page, right?

**Comment by gaow** [comment]
> After the pipeline has been executed, the user can start a workflowr analysis to run arbitrary complex queries directly with dsc_query and then immediately manipulate and visualize the results.

Exactly. It is more like post-dsc analysis with an official guideline.

**Comment by pcarbo** [comment]
@jdblischak @gaow It sounds very reasonable to demonstrate the use of workflowr together with dsc2; in fact, this combination seems quite natural---dsc2 to generate the results, R Markdown files for exploring and summarizing the results. 

I don't see workflowr as a "dependency"; this is just an illustration of workflowr in a specific setting (a DSC experiment).
